common/dal2/src/services/fs.rs

```diff
@@ -90,8 +100,23 @@ impl Accessor for Backend {
     }
 
     async fn write(&self, mut r: Reader, args: &OpWrite) -> Result<usize> {
+        println!("try to write {}", args.path);
+
         let path = PathBuf::from(&self.root).join(&args.path);
 
+        // Create dir before write path.
+        //
+        // TODO(xuanwo): There are many works to do here:
+        //   - Is it safe to create dir concurrently?
+        //   - Do we need to extract this logic as new util functions?
+        //   - Is it better to check the parent dir exists before call mkdir?
+        let parent = path
```

Comment:
Does this only for `Fs` storage type?
We already inited the dir in:
https://github.com/datafuselabs/databend/blob/5bf157ed57e20566b99cc65ed296f2719175b121/query/src/bin/databend-query.rs#L49-L50

Reply:
I have removed `conf.initial_dir()` logic. As I think it's better to let DAL handle those work.

Reply:
Oh, I just found that this comment is on the `write` operation, please allow me to explain a bit more.

`Write` in DAL supports `write("a/b/c/d/e/f")` so the caller doesn't bother to call `mkdir` alike APIs. So we must create dir for `parent` here.